config/v3_1/config.go

```diff
@@ -56,6 +59,28 @@ func Parse(rawConfig []byte) (types.Config, report.Report, error) {
 	return config, rpt, nil
 }
 
-func isEmpty(userdata []byte) bool {
-	return len(userdata) == 0
+// ParseConfig parses the raw config of version 3.1.0 or lesser
+// into a 3.1 types.Config struct and generates a report of any
+// errors, warnings,info, and deprecations it encountered
+func ParseConfig(raw []byte) (types.Config, report.Report, error) {
+	version, rpt, err := util.GetConfigVersion(raw)
+	if err != nil {
+		return types.Config{}, rpt, err
+	}
+
+	switch version {
+	case types.MaxVersion:
+		return Parse(raw)
+	case types_3_0.MaxVersion:
+		return v3_1_from_3_0(v3_0.ParseConfig(raw))
+	default:
+		return types.Config{}, report.Report{}, errors.ErrUnknownVersion
+	}
+}
+
+func v3_1_from_3_0(cfg types_3_0.Config, r report.Report, err error) (types.Config, report.Report, error) {
+	if err != nil {
```

Comment:
These functions are only used once each, so optionally their code could be inlined into their caller.